src/plugins/data/common/index_patterns/fields/field.ts

```diff
@@ -148,6 +146,26 @@ export class Field implements IFieldType {
     // multi info
     obj.fact('subType');
 
-    return obj.create();
+    const newObj = obj.create();
+    newObj.toSpec = function () {
+      return {
+        count: this.count,
+        script: this.script,
+        lang: this.lang,
+        conflictDescriptions: this.conflictDescriptions,
+        name: this.name,
+        type: this.type,
+        esTypes: this.esTypes,
+        scripted: this.scripted,
+        searchable: this.searchable,
+        aggregatable: this.aggregatable,
+        readFromDocValues: this.readFromDocValues,
+        subType: this.subType,
+        format: this?.indexPattern?.fieldFormatMap[this.name]?.toJSON(),
```

Comment:
Generally speaking, the code related to saved objects (my starting point for this work) omits fields that don't have values but I'm happy to mirror how its done elsewhere.

Reply:
Typically I'd advocate for `undefined` since I think it is clearer than returning some default empty state, so I could go either way on this as well. I only mentioned it because I had seen it done differently elsewhere.